packages/jest-each/src/bind.js

```diff
@@ -34,16 +34,18 @@ export default (cb: Function) => (...args: any) => (
   const table = buildTable(data, keys.length, keys);
 
   if (data.length % keys.length !== 0) {
+    const error = new Error(
+      'Not enough arguments supplied for given headings:\n' +
+        EXPECTED_COLOR(keys.join(' | ')) +
+        '\n\n' +
+        'Received:\n' +
+        RECEIVED_COLOR(pretty(data)) +
+        '\n\n' +
+        `Missing ${RECEIVED_COLOR(`${data.length % keys.length}`)} arguments`,
+    );
+
     return cb(title, () => {
```

Comment:
@mattphillips the idea is to create the error within the same async tick as the method call from the user - the moment you cross an async barrier (like executing the body of a test) the original stack is lost

Reply:
throwing a string instead of a new error would probably work as well, as we'd discover there was no stack, and use the one we keep around for async errors. Throwing an error though gives us a stack - it's just not usable